### PR TITLE
[4.0] Rewrite sampledata-process.js

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #19263

### Summary of Changes

- Rewrite in vanilla JS
- Use Custom Element alerts instead of BS
- Progress bar now increments as per the step it's on.
- Create minified file

### Testing Instructions

If you want to test, you'll need to ensure you have no sample data installed already. So install a fresh copy of Joomla and attempt to install sample data.

If you **do** have sample data already installed, the process should stop on step 1 and display an error alert.

### CODE REVIEW

**Review by dgrammatiko — APPROVED** [review]

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on 22602685a98f3512fa07841c399b7c5c819bde73<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19269">issues.joomla.org/tracker/joomla-cms/19269</a>.</sub>

**Comment by brianteeman** [comment]
~Maybe its just me but it seems wrong to have a class on a list called list-unstyled~

Sorry realised that its an actual class in bootstrap - crazy

File: administrator/modules/mod_sampledata/tmpl/default.php [hunk 1]
```diff
@@ -35,20 +38,20 @@
 						<span class="icon-<?php echo $item->icon; ?>" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/modules/mod_sampledata/tmpl/default.php]: Replace with `fa`

**Comment by laoneo** [comment]
Let us know when we can test.

**Comment by C-Lodder** [comment]
@laoneo - ready now

**Comment by dgrammatiko** [comment]
@C-Lodder couple of things that you need to do here:
- change the ajax method from GET to POST
- revert the `data` part of the ajax method
- change the `$app->input` to `$app->input->json` in the php, components/com_ajax/ajax.php:

```php
// Check for valid format
if (!$format)
{
	// JInput JSON object
	$input = $input = $app->input->json;

	// Requested format passed via URL
	$format = strtolower($input->getWord('format'));

	if (!$format) {
		$results = new InvalidArgumentException(JText::_('COM_AJAX_SPECIFY_FORMAT'), 404);
	}
}
```

File: administrator/modules/mod_sampledata/tmpl/default.php [hunk 2]
```diff
@@ -14,41 +14,44 @@
...
 		<?php foreach($items as $i => $item) : ?>
 			<li class="list-group-item sampledata-<?php echo $item->name; ?>">
 				<div class="d-flex justify-content-between align-items-center">
 					<div class="mr-2">
-						<span class="icon-<?php echo $item->icon; ?>" aria-hidden="true"></span>
+						<span class="fa fa-<?php echo $item->icon; ?>" aria-hidden="true"></span>
 						<?php echo htmlspecialchars($item->title, ENT_QUOTES, 'UTF-8'); ?>
 					</div>
-					<a href="#" class="btn btn-primary btn-sm" onclick="sampledataApply(this)" data-type="<?php echo $item->name; ?>" data-steps="<?php echo $item->steps; ?>">
+					<a href="#" class="btn btn-primary btn-sm" onclick="Joomla.sampledataApply(this)" data-type="<?php echo $item->name; ?>" data-steps="<?php echo $item->steps; ?>">
```
Comment (dgrammatiko) [inline administrator/modules/mod_sampledata/tmpl/default.php]: @C-Lodder can you remove the `onclick` here and add an event listener in the js file. CSP strict...

Comment (C-Lodder) [reply administrator/modules/mod_sampledata/tmpl/default.php]: sure, will do

**Comment by C-Lodder** [comment]
@dgt41 - Inline script removed. Didn't you say `com_ajax` didn't support json?

**Comment by dgrammatiko** [comment]
@C-Lodder I haven't actually tested the snippet above, so it might or might not work. Anyways this is pretty fine to be merged as is (if you don't have the time to dig into switching to json post)

**Comment by C-Lodder** [comment]
@laoneo @Anu1601CS  ready to test

**Comment by C-Lodder** [comment]
Closing in favour of: https://github.com/joomla-projects/joomla-es6/pull/58

**Pull request closed without merging**